packages/components/src/components/post-megadropdown/post-megadropdown.tsx

```diff
@@ -131,32 +175,52 @@ export class PostMegadropdown {
     this.firstFocusableEl?.focus();
   }
 
-  private breakpointChange(e: CustomEvent) {
-    this.device = e.detail;
-    if (this.device === 'desktop' && this.isVisible) {
-      this.animationClass = null;
-    }
-  }
   /**
    * Forces the dropdown to close without animation.
    */
   private forceClose() {
-    this.isVisible = false;
-    this.animationClass = null;
-    this.postToggleMegadropdown.emit({ isVisible: this.isVisible, focusParent: false });
-    this.removeListeners();
+    this.cancelAllAnimations();
+    this.closeCleanUp();
+    this.postToggleMegadropdown.emit({ isVisible: false, focusParent: false });
   }
 
-  private handleAnimationEnd() {
-    if (this.animationClass === 'slide-out') {
-      this.isVisible = false;
-      this.animationClass = null;
-      PostMegadropdown.activeDropdown = null;
-      this.removeListeners();
+  // Run the respective animation
+  private createAnimation(direction: 'in' | 'out'): Animation {
+    if (this.device === 'desktop') {
+      return fadeSlide(this.animatedContainer, direction, this.fsAnimationOptions);
     }
+    return slide(this.animatedContainer, direction, {
+      translate: 100,
+      duration: 350,
+      easing: direction === 'in' ? 'ease-in' : 'ease-out',
+    });
+  }
+
+  private cancelAllAnimations() {
+    this.currentAnimation?.cancel();
+    this.animatedContainer?.getAnimations().forEach(a => a.cancel());
+    this.currentAnimation = null;
   }
 
-  private readonly handleClickOutside = (event: MouseEvent) => {
+  // Clean up and reset state to non visible
+  private closeCleanUp() {
+    this.resetAnimationState();
+    this.removeListeners();
+    this.isVisible = false;
+    this.postToggleMegadropdown.emit({ isVisible: this.isVisible });
```

Comment:
This event is emitted every time the drop-down opens or closes, on top of the events emitted in the `hide` and `show` methods.